dynamic-property-spring/src/main/java/ru/fix/dynamic/property/spring/DynamicPropertyAwareBeanPostProcessor.java

```diff
@@ -0,0 +1,129 @@
+        processingTime.add(currentProcessingTime);
+        log.debug("Resolving @PropertyId annotation for '{}' bean took {} ms. " +
+                        "Sum of processing times is equal {} ms now.",
+                beanName, currentProcessingTime, processingTime.sum()
+        );
+
+        return bean;
+    }
+
+    private Object processDynamicProperty(Object bean, Field field, PropertyId propertyIdAnnotation, String beanName) {
+
+        Class<?> fieldType = field.getType();
+
+        if (fieldType.isAssignableFrom(DynamicProperty.class)) {
+
+            String propertyId = propertyIdAnnotation.value();
+
+            Class propertyClass = extractPropertyClass(field);
+            Object propertyDefaultValue = extractDefaultValue(bean, field);
+            if (null == propertyDefaultValue) {
+                String errorMessage = String.format(
+                        "Illegal default property value '%s' of bean '%s'. " +
+                                "DynamicProperty type annotated by @PropertyId must have default value other than null",
+                        field.getName(), beanName
+                );
+                throw new DynamicPropertyDefaultValueNotFoundException(errorMessage);
+            }
+
+            addDefaultValueIfAbsent(propertyId, propertyDefaultValue);
+
+            //noinspection unchecked
+            return new SourcedProperty<>(propertySource, propertyId, propertyClass, propertyDefaultValue);
+
+        } else {
+            log.warn(
+                    "@PropertyId annotation is applicable only on fields " +
+                            "of DynamicProperty type, not '{}', bean '{}'",
+                    fieldType, beanName
+            );
+            return null;
+        }
+    }
+
+    private Class extractPropertyClass(Field field) {
+        ParameterizedType parameterizedType = (ParameterizedType) field.getGenericType();
+        Type type = parameterizedType.getActualTypeArguments()[0];
+        Class propertyClass;
+        if (type instanceof ParameterizedType) {
+            propertyClass = (Class) ((ParameterizedType) type).getRawType();
+        } else {
+            propertyClass = (Class) type;
+        }
+        return propertyClass;
+    }
+
+    private Object extractDefaultValue(Object bean, Field field) {
+        DynamicProperty<?> dynamicProperty = null;
+        try {
+            dynamicProperty = (DynamicProperty<?>) field.get(bean);
+        } catch (IllegalAccessException e) {
+            log.error("Error occurred when extracting value from field '{}'", field.getName());
+        }
+        return Optional.ofNullable(dynamicProperty)
+                .map(DynamicProperty::get)
+                .orElse(null);
+    }
+
+    private void addDefaultValueIfAbsent(String propertyId, Object propertyDefaultValue) {
+        try {
+            propertySource.putIfAbsent(propertyId, propertyDefaultValue);
```

Comment:
We are not pushing default value from code to PropertySource, no need in this method

Reply:
fix